framework/src/material_image_helpers.cpp

```diff
@@ -284,6 +421,63 @@ namespace gvk
 
 		return std::make_tuple(std::move(positionsBuffer), std::move(indexBuffer));
 	}
+	
+	std::tuple<avk::buffer, avk::buffer> create_vertex_and_index_buffers(const std::vector<std::tuple<avk::resource_reference<const gvk::model_t>, std::vector<size_t>>>& aModelsAndSelectedMeshes, vk::BufferUsageFlags aUsageFlags, avk::sync aSyncHandler)
+	{
+		return create_vertex_and_index_buffers(get_vertices_and_indices(aModelsAndSelectedMeshes),
+			aUsageFlags, std::move(aSyncHandler));
+	}
+
+	std::tuple<avk::buffer, avk::buffer> create_vertex_and_index_buffers_cached(const std::vector<std::tuple<avk::resource_reference<const gvk::model_t>, std::vector<size_t>>>& aModelsAndSelectedMeshes, vk::BufferUsageFlags aUsageFlags, avk::sync aSyncHandler, gvk::serializer& aSerializer)
+	{
+		size_t numPositions = 0;
+		size_t totalPositionsSize = 0;
+		size_t numIndices = 0;
+		size_t totalIndicesSize = 0;
+
+		if (aSerializer.mode() == gvk::serializer::mode::serialize) {
+			auto [positionsData, indicesData] = get_vertices_and_indices(aModelsAndSelectedMeshes);
+
+			numPositions = positionsData.size();
+			totalPositionsSize = sizeof(positionsData[0]) * numPositions;
+			numIndices = indicesData.size();
+			totalIndicesSize = sizeof(indicesData[0]) * numIndices;
+
+			aSerializer.archive(numPositions);
+			aSerializer.archive(totalPositionsSize);
+			aSerializer.archive(numIndices);
+			aSerializer.archive(totalIndicesSize);
+
+			aSerializer.archive(aSerializer.binary_data(positionsData.data(), totalPositionsSize));
+			aSerializer.archive(aSerializer.binary_data(indicesData.data(), totalIndicesSize));
+
+			return create_vertex_and_index_buffers(std::make_tuple(positionsData, indicesData), aUsageFlags, std::move(aSyncHandler));
```

Comment:
This should be `std::make_tuple(std::move(positionsData), std::move(indicesData))`, otherwise unnecessary copies of the `std::vector`s are made.